Dist Updates for PR Challenge

genio opened the pull request:
Hi!

I received your dist as part of this month's CPAN [Pull Request Challenge](http://cpan-prc.org/).

I made almost zero actual coding changes to your dist.  So, while this seems like a rather large PR (for which I apologize) there is next to nothing in terms of actual code changes.

Code changes:
- Got rid of indirect-object-syntax.  I changed `new Foo` to `Foo->new` and `import Foo` to `Foo->import`.
- I standardized your `$VERSION` declarations across your `.PM` files, adhering to [Version Numbers Should be Boring](http://www.dagolden.com/index.php/369/version-numbers-should-be-boring/).

Test Suite:
- Added some boiler-plate tests in `xt/author` and `xt/release` that are solely for you, as the developer, to run and make sure your dist is in as good of shape as it can be. `prove -lr xt` will run these for you.
- Added a little boiler-plate testing in `t`.  I also moved your test files out of there and into `xt/author` since they weren't mentioned in the `MANIFEST` file before, they weren't being run, so I figured you wanted it that way.
- Moved `test.pl` to `t/02-coms.t`.  This works just as it did before, but now uses `Test::More` instead of rolling your own testing suite.
- Made all tests make use of `Test::More`.
  - `perl Makefile.PL`
  - `make test` OR `prove -lr t` (neither of these will run tests in the `xt/` dir)
  - `prove -lr xt` for optional tests

Build:
- Updated the `Makefile.PL` a bit to include the prereqs that were missing before.
- Updated the `MANIFEST` to include all of the files in the distribution
- Updated the `.gitignore` to no longer worry about build files

cosimo commented:
This is an awesome piece of work. Thank you!
I apologize that it took me so long to get to it, but I never saw this before today...

cosimo commented on Makefile.PL:
```diff
@@ -1,46 +1,101 @@
+  "CONFIGURE_REQUIRES" => {
+    "ExtUtils::MakeMaker" => 0
+  },
+  "DISTNAME" => "Mail-Sender",
```
Sure this is correct? Shouldn't it be `Device-Modem`?

cosimo commented on t/01-load.t:
```diff
@@ -0,0 +1,15 @@
+use warnings;
+use Test::More tests => 6;
+
+BEGIN {
```
I assumed the use of `use_ok()` was deprecated a long time ago. See http://blogs.perl.org/users/ovid/2012/04/avoiding-use-ok-in-t00-loadt.html for example. Is it considered ok?

cosimo commented:
I've added a few comments to the relevant sections of the code, but want to thank you again for this work. Very appreciated.
Tests are failing, but just because travis hasn't been configured. Would you be willing to add a default `.travis-ci.yml` in place so as to make the test suite run correctly? That would be great.

genio commented:
Oops.  I fixed the copy-pasta error on the Makefile.  I added the Travis config and it appears to be testing correctly.

As for the load.t, I really don't have a strong argument for or against.  I mean, the compile tests are there in xt/author, so I could remove that test completely?

cosimo commented:
Very thorough improvement work. Thank you very much!
